packages/app/src/cli/services/payments/extension-to-toml.ts

```diff
@@ -0,0 +1,110 @@
+import {ExtensionRegistration} from '../../api/graphql/all_app_extension_registrations.js'
+import {BasePaymentsAppExtensionDeployConfigType} from '../../models/extensions/specifications/payments_app_extension_schemas/base_payments_app_extension_schema.js'
+import {
+  RedeemablePaymentsAppExtensionDeployConfigType,
+  redeemableDeployConfigToCLIConfig,
+} from '../../models/extensions/specifications/payments_app_extension_schemas/redeemable_payments_app_extension_schema.js'
+import {
+  OffsitePaymentsAppExtensionDeployConfigType,
+  offsiteDeployConfigToCLIConfig,
+} from '../../models/extensions/specifications/payments_app_extension_schemas/offsite_payments_app_extension_schema.js'
+import {
+  CustomCreditCardPaymentsAppExtensionDeployConfigType,
+  customCreditCardDeployConfigToCLIConfig,
+} from '../../models/extensions/specifications/payments_app_extension_schemas/custom_credit_card_payments_app_extension_schema.js'
+import {
+  CreditCardPaymentsAppExtensionDeployConfigType,
+  creditCardDeployConfigToCLIConfig,
+} from '../../models/extensions/specifications/payments_app_extension_schemas/credit_card_payments_app_extension_schema.js'
+import {
+  CustomOnsitePaymentsAppExtensionDeployConfigType,
+  customOnsiteDeployConfigToCLIConfig,
+} from '../../models/extensions/specifications/payments_app_extension_schemas/custom_onsite_payments_app_extension_schema.js'
+import {encodeToml} from '@shopify/cli-kit/node/toml'
+import {slugify} from '@shopify/cli-kit/common/string'
+
+function typeToContext(type: string) {
+  switch (type) {
+    case DashboardPaymentExtensionType.Offsite:
+      return 'offsite'
+    case DashboardPaymentExtensionType.CreditCard:
+      return 'credit-card'
+    case DashboardPaymentExtensionType.CustomCreditCard:
+      return 'custom-credit-card'
+    case DashboardPaymentExtensionType.CustomOnsite:
+      return 'custom-onsite'
+    case DashboardPaymentExtensionType.Redeemable:
+      return 'redeemable'
+  }
+}
+
+export enum DashboardPaymentExtensionType {
+  Offsite = 'payments_app',
+  CreditCard = 'payments_app_credit_card',
+  CustomCreditCard = 'payments_app_custom_credit_card',
+  CustomOnsite = 'payments_app_custom_onsite',
+  Redeemable = 'payments_app_redeemable',
+}
+
+export async function buildTomlObject(extension: ExtensionRegistration) {
```

Comment:
I don't see why this function needs to be `async` 🤔

Reply:
Fixed, thanks